intake_esm/core.py

```diff
@@ -756,7 +756,10 @@ def _is_pattern(value):
     if value_is_repattern:
         return True
     wildcard_chars = {'*', '?', '$', '^'}
-    return any(char in value for char in wildcard_chars)
+    try:
+        return any(char in value for char in wildcard_chars)
+    except TypeError:
+        return False
```

Comment:
This does not test for escaped character sequences.  Presumably this is not a huge problem right now, but what if someone has a filename with a `$` in it?  Or a `*`?  Would it be safe to replace this test with:

```suggestion
    try:
        value_ = value
        for char in wildcard_chars:
            value_ = value_.replace(f'\{char}', '')
        return any(char in value_ for char in wildcard_chars)
    except TypeError:
        return False
```